bob-apps/bin/bobp.rs

```diff
@@ -470,113 +592,117 @@ async fn stat_worker(
         print_periodic_stat(stop_token, period_ms, &stat, request_bytes, behavior_flags);
-    print_averages(
-        &stat,
-        &put_speed_values,
-        &get_speed_values,
-        &exist_speed_values,
-        elapsed,
-        keys_count,
-        behavior_flags,
-    );
-    print_errors_with_codes(stat);
+    print_averages(&stat, &put_speed_values, &get_speed_values, &exist_speed_values, &delete_speed_values, elapsed, keys_count, behavior_flags);
+    print_errors_with_codes(stat).await
 }
 
-fn print_errors_with_codes(stat: Arc<Statistics>) {
-    let guard = stat.get_errors.lock().expect("mutex");
+async fn print_errors_with_codes(stat: Arc<Statistics>) {
+    let guard = stat.get.common.errors.lock().expect("mutex");
     if !guard.is_empty() {
         println!("get errors:");
         for (&code, count) in guard.iter() {
             println!("{:?} = {}", Code::from(code), count);
         }
     }
-    let guard = stat.put_errors.lock().expect("mutex");
+    let guard = stat.put.common.errors.lock().expect("mutex");
     if !guard.is_empty() {
         println!("put errors:");
         for (&code, count) in guard.iter() {
             println!("{:?} = {}", Code::from(code), count);
         }
     }
-    let guard = stat.exist_errors.lock().expect("mutex");
+    let guard = stat.exist.common.errors.lock().expect("mutex");
     if !guard.is_empty() {
         println!("exist errors:");
         for (&code, count) in guard.iter() {
             println!("{:?} = {}", Code::from(code), count);
         }
     }
+    let guard = stat.delete.common.errors.lock().expect("mutex");
+    if !guard.is_empty() {
+        println!("delete errors:");
+        for (&code, count) in guard.iter() {
+            println!("{:?} = {}", Code::from(code), count);
+        }
+    }
 }
 
 fn print_averages(
     stat: &Arc<Statistics>,
     put_speed_values: &[f64],
     get_speed_values: &[f64],
     exist_speed_values: &[f64],
+    delete_speed_values: &[f64],
     elapsed: Duration,
     keys_count: u64,
     behavior_flags: u8,
 ) {
-    let total_put = stat.put_total.load(Ordering::Relaxed);
-    let total_exist = stat.exist_total.load(Ordering::Relaxed);
-    let total_get = stat.get_total.load(Ordering::Relaxed);
-    let avg_total = ((total_put + total_get + total_exist) * 1000)
-        .checked_div(elapsed.as_millis() as u64)
-        .unwrap_or_default();
-    let total_err = stat.put_error_count.load(Ordering::Relaxed)
-        + stat.get_error_count.load(Ordering::Relaxed)
-        + stat.exist_error_count.load(Ordering::Relaxed);
+    let total_put = stat.put.common.total.load(Ordering::Relaxed);
+    let total_exist = stat.exist.common.total.load(Ordering::Relaxed);
+    let total_get = stat.get.common.total.load(Ordering::Relaxed);
+    let total_delete = stat.delete.common.total.load(Ordering::Relaxed);
+    let avg_total = ((total_put + total_get + total_exist + total_delete) * 1000)
```

Comment:
Add `get_total` function to `Statistics` that calculates the sum of `total` values